systemtests improvements - speed up test setup & cleanup on Vagrant

unclejack opened the pull request:
This PR speeds up the setup and tear down of the tests. These changes reduce the execution of the system tests by about  30-40 minutes for the regular test suite.

unclejack commented:
This PR is ready for review.

unclejack commented:
@abhinandanpb @jojimt @rhim Can you take a look at this PR, please? We'll save quite a bit of time on the CI. I've tested this with the old Vagrant box and with the new Vagrant box.

rhim commented:
LGTM.
- Himanshu
     .:|:.:|:.

On Mar 1, 2017, at 2:12 AM, unclejack <notifications@github.com<mailto:notifications@github.com>> wrote:


@abhinandanpb<https://github.com/abhinandanpb> @jojimt<https://github.com/jojimt> @rhim<https://github.com/rhim> Can you take a look at this PR, please? We'll save quite a bit of time on the CI. I've tested this with the old Vagrant box and with the new Vagrant box.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/contiv/netplugin/pull/752#issuecomment-283299396>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AMiMKQlyu_CK6XApmdxSLop6XdADvr8vks5rhURxgaJpZM4MHajl>.

jojimt commented on test/systemtests/util_test.go:
```diff
@@ -1328,36 +1367,50 @@ func (s *systemtestSuite) SetUpTestBaremetal(c *C) {
 	}
 
-	time.Sleep(15 * time.Second)
+	time.Sleep(5 * time.Second)
```
I don't think we should remove these delays unless we add a polling to test for the status we are expecting. These delays were added to address failures due to timing differences between machines.

unclejack replied:
You're right. I'll add these back. Do you happen to have some suggestions related to what we should poll?

unclejack commented:
@jojimt: I've changed back those time.Sleep to their original values. I've also rebased to the current master.

The CI should be green in a few hours.

PTAL

unclejack commented:
@jojimt: This is green. PTAL
